### PR TITLE
feat(select): integrate with Angular forms

### PR DESCRIPTION
**Pull request by kara**
This PR adds basic forms integration. 
Still TODO for forms:
- disabled support (see #1667)

r: @jelbourn

### CODE REVIEW

File: src/lib/select/option.ts [hunk 1]
```diff
@@ -46,9 +50,9 @@ export class MdOption {
   }
 
   /** Selects the option. */
-  select(): void {
+  select(isUserInput = false): void {
```
Comment (kara) [inline src/lib/select/option.ts]: That makes way more sense. Good call.

File: src/lib/select/select.spec.ts [hunk 2]
```diff
@@ -206,6 +207,102 @@ describe('MdSelect', () => {
...
+    beforeEach(() => {
+      fixture = TestBed.createComponent(BasicSelect);
+      fixture.detectChanges();
+
+      trigger = fixture.debugElement.query(By.css('.md-select-trigger')).nativeElement;
+    });
+
+    it('should set the view value from the form', () => {
+      let value = fixture.debugElement.query(By.css('.md-select-value'));
+      expect(value).toBeNull();
```
Comment (kara) [inline src/lib/select/select.spec.ts]: Ack, I always forget about that

File: src/lib/select/select.ts [hunk 3]
```diff
@@ -110,6 +136,38 @@ export class MdSelect implements AfterContentInit, OnDestroy {
...
   }
 
+  /**
+   * Sets the select's value. Part of the ControlValueAccessor interface
+   * required to integrate with Angular's core forms API.
+   */
+  writeValue(value: any): void {
+    if (!this.options) { return; }
+
+    this.options.forEach((option: MdOption) => {
```
Comment (kara) [inline src/lib/select/select.ts]: this.options is a QueryList, so you'd have to use toArray() first.

**Comment by kara** [comment]
@jelbourn Comments addressed

File: src/lib/select/select.spec.ts [hunk 4]
```diff
@@ -206,6 +207,114 @@ describe('MdSelect', () => {
...
+
+    it('should set the view value from the form', () => {
+      let value = fixture.debugElement.query(By.css('.md-select-value'));
+      expect(value).toBeNull('Expected trigger to start with empty value.');
+
+      fixture.componentInstance.control.setValue('pizza-1');
+      fixture.detectChanges();
+
+      value = fixture.debugElement.query(By.css('.md-select-value'));
+      expect(value.nativeElement.textContent).toContain('Pizza',
```
Comment (jelbourn) [inline src/lib/select/select.spec.ts]: Prefer breaking at the higher syntactic level, e.g.:
```ts
expect(value.nativeElement.textContent)
    .toContain('Pizza', `Expected trigger to be populated by the control's new value.`);
```
(general rule-of-thumb that makes code more readable in all languages)

Comment (kara) [inline src/lib/select/select.spec.ts]: Yeah, that does look better. I'll reformat.

**Pull request merged**